tcp_check/tests/test_tcp_check.py

```diff
@@ -93,11 +94,42 @@ def test_response_time(aggregator):
 
     # service check
     expected_tags = ['foo:bar', 'target_host:datadoghq.com', 'port:80', 'instance:instance:response_time']
+    expected_tags.append("address:{}".format(check._addrs[0]))
     aggregator.assert_service_check('tcp.can_connect', status=check.OK, tags=expected_tags)
     aggregator.assert_metric('network.tcp.can_connect', value=1, tags=expected_tags)
 
     # response time metric
     expected_tags = ['url:datadoghq.com:80', 'instance:instance:response_time', 'foo:bar']
+    expected_tags.append("address:{}".format(check._addrs[0]))
     aggregator.assert_metric('network.tcp.response_time', tags=expected_tags)
     aggregator.assert_all_metrics_covered()
     assert len(aggregator.service_checks('tcp.can_connect')) == 1
+
+
+def test_multiple(aggregator):
+    """
+    Test when a domain is attached to 3 IPs [UP, DOWN, UP]
+    """
+    instance = deepcopy(common.INSTANCE_MULTIPLE)
+    instance['name'] = 'multiple'
+    instance['ip_cache_duration'] = 0
+    check = TCPCheck(common.CHECK_NAME, {}, [instance])
+
+    with mock.patch('socket.gethostbyname_ex', return_value=[None, None, ['ip1', 'ip2', 'ip3']]), mock.patch.object(
+        check, 'connect', wraps=check.connect
+    ) as connect:
+        connect.side_effect = [None, Exception(), None] * 2
+        expected_tags = ['foo:bar', 'target_host:datadoghq.com', 'port:80', 'instance:multiple']
+
+        # Running the check twice
+        check.check(None)
+        check.check(None)
```

Comment:
```suggestion
        dd_run_check(check)
        dd_run_check(check)
```
small nit

Reply:
I saw that `tcp_check` still uses `check.check(None)` for its other non-e2e tests, so it's up to you if you want to apply this change or keep things consistent for a separate PR.

Reply:
Let's do this in another PR